src/routes/webhooks/apple.ts

```diff
@@ -98,20 +173,72 @@ export const apple = async (fastify: FastifyInstance): Promise<void> => {
         const notification =
           await verifier.verifyAndDecodeNotification(signedPayload);
 
+        if (isNullOrUndefined(notification?.data?.signedRenewalInfo)) {
+          logger.info(
+            { notification },
+            "[Apple StoreKit]: Missing 'signedRenewalInfo' in notification data",
+          );
+          return response.status(400).send({ error: 'Invalid Payload' });
+        }
+
+        const renewalInfo = await verifier.verifyAndDecodeRenewalInfo(
+          notification.data.signedRenewalInfo!,
+        );
+
+        const con = await createOrGetConnection();
+        const user = await con.getRepository(User).findOne({
+          select: ['id', 'subscriptionFlags'],
+          where: {
+            subscriptionFlags: JsonContains({
+              appAccountToken: renewalInfo.appAccountToken,
+            }),
+          },
+        });
+
+        if (!user) {
+          logger.error(
+            { notification },
+            '[Apple StoreKit]: User not found with matching app account token',
+          );
+          return response.status(404).send({ error: 'Invalid Payload' });
+        }
+
         logger.info(
-          { notification },
-          'Received Apple App Store Server Notification',
+          { renewalInfo, user },
+          '[Apple StoreKit]: Received Apple App Store Server Notification',
         );
+
+        const subscriptionStatus = getSubscriptionStatus(
+          notification.notificationType,
+          notification.subtype,
+        );
+
+        const subscriptionFlags = renewalInfoToSubscriptionFlags(renewalInfo);
+
+        await updateStoreKitUserSubscription({
```

Comment:
One thing that comes to mind, can we catch if sub already exists on paddle for some reason and throw error in that case so we are aware of it?

Reply:
Same thing would be helpful in inverse case where someone has apple sub but then tries to purchase paddle

Reply:
just so we don't have overwriting subs

Reply:
Yeah, I've been thinking about that. Not 100% sure what the best course of action is. I think it's better to let StoreKit overwrite, as we can't issue refunds on behalf of the user, but we can on Paddle.

But of course log when it happens, on both sides.

Reply:
I would say throw, and handle on case by case basis manually, we have restrictions on front side so technically it should not be visible, but let's raise it on the channel to see what Ido/product think?

Reply:
It now throws, and I'll add some log alerts so we get notified on the cases. Yeah, shouldn't happen, but who knows 🙈